File: VehicleInsuranceApp/app/src/main/java/com/example/vehicleinsuranceapp/MainActivity.java
package com.example.vehicleinsuranceapp;

import android.os.Bundle;
import android.widget.FrameLayout;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.example.vehicleinsuranceapp.Fragment.LoginFragment;

public class MainActivity extends AppCompatActivity {

    FrameLayout framelayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        framelayout = (FrameLayout)findViewById(R.id.frame_layout);// navigate to act to frag
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.add(R.id.frame_layout,new LoginFragment()).commit();
    }
}

File: android kotlin/SmartHomeControllerNew/app/src/main/java/com/example/smarthomecontroller/MainActivity.kt
package com.example.smarthomecontroller

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import com.example.smarthomecontroller.MainActivity

class MainActivity : AppCompatActivity() {

    private val mainViewModel: MainViewModel by viewModels()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val textViewStatus = findViewById<TextView>(R.id.textViewStatus)
        mainViewModel.deviceStatus.observe(this, Observer { status ->
            textViewStatus.text = status
        })

        mainViewModel.fetchDeviceStatus()
            mainViewModel.fetchDeviceStatus()
        }
    }

